scala-package/core/src/main/scala/org/apache/mxnet/io/NDArrayIter.scala

```diff
@@ -101,10 +115,16 @@ class NDArrayIter(data: IndexedSeq[(String, NDArray)],
   private var cursor = -dataBatchSize
 
   private val (_provideData: ListMap[String, Shape],
-               _provideLabel: ListMap[String, Shape]) = {
+               _provideLabel: ListMap[String, Shape],
```

Comment:
can you separate the _provideDataDesc and _provideLabelDesc into separate statement

Reply:
@yzhliu recommend do this several days ago in previous CR. Which one works better?

Reply:
My major point was no need to fetch the data twice